### ISSUE TITLE
Which calibrator catalogue to use?

### ISSUE THREAD

**Issue by rubyvanrooyen** [rubyvanrooyen/astrokat#5]
If we are to provide calibration standards to the users for selection we need to discuss what to use and where this goes.

We need to have a standard calibrator catalogue (for L-band).

- Which catalogues do we have?
Last I know Tony was supposed to generate observatory guaranteed calibrator lists to be added to the katconfig/users/catalogue repository, but there are bunches of branches and an outdated master?

- Shall we use katsdpcatalogues? This is not advised since things is used by many users.

- Can commissioning follow this up and make a decision, implementation will rely on it. It does not have to be permanent, but need to be documented .

**Comment by mserylak** [comment]
There is at least 2 places where catalogues sit for imaging (you mention them above) and at least 3 where the targets are specified for beamforming (that I know of).

- katsdpcatalogues,
- katconfig,
- on PTUSE,

PTUSE is not in our discretion to worry about. I know it will be a problem when PTUSE will be used in open time observations (as a service instrument) and users might want to specify a target to observe (and fold) and the ephemeris will not be there. This will make observation to fail. MeerTIME is working on a solution now.

In general it is a bit of a mission to orient yourself in branches etc. I would say that standard list of observatory supported targets should be done via katconfig. I think katsdpcatalogues should be used as experimental/testing/commissioning bucket. I might be wrong and commissioning will assume that katsdpcatalogues should be the place to put catalogues from external users... @spassmoor any thoughts?

**Comment by spassmoor** [comment]
I think the choice of calibrator should be made by the person submitting the observation config file and there should not be any catalogue files in the system.  

> We need to have a standard calibrator catalogue (for L-band)


This would be needed for the OPT or PMT  but not for this scripting project. Since the expert/astronomer would have chosen the calibrator then put it into the config file.

**Comment by rubyvanrooyen** [comment]
I think you may be confusing our assistance to CAM during their tools development with our observation script development. These 2 attempts are currently running in parallel, but have not merged yet.

I agree that the catalogue submission should be phased out and that the observation config file will take over.

My question is however related to a very practical issue: CAM needs to build a tool for COMM that will allow them to view/select/suggest calibrators to astronomers. They need to use a standard catalogue/list/'_insert your favourite word here_'

I had assumed we had agreed during AR1 already that these types of calibrator options will live in katconfig. Currently this is outdated with a myriad of branches from Tony, but nothing merged into main or karoo.

There are also catalogues that claim to be calibrator lists in katsdpcatalogues, which should be cleared out or used.

This is confusing for implementation.

What does COMM suggest? Asking that Tony merge those branches with calibrators that can be used in katconfig or use something from katsdpcatalogues. You cannot have both and I really would like to drop katsdpcatalogues.

**Comment by spassmoor** [comment]
Yes it appears I have confused the issue with my comment. The CAM helper tool to suggest calibrators would be : 

> import katpoint
> import numpy as np
> ant = katpoint.Antenna('ant1, -30:42:39.8, 21:26:38.0, 1035.0') # the MeerKAT reference point
> ant.ref_observer.horizon = '20:00:00' # horizon set to 20 degrees
> cat = katpoint.Catalogue(file('/var/kat/katsdpcatalogues/bpcals.csv'),antenna=ant)
> target = katpoint.Target('0252-712,radec gaincal,02:52:46.157,-71:04:35.28') # Add in target 
> for pot_cal in cat.iterfilter(dist_limit_deg=[1.,15.],proximity_targets=target,antenna=ant) :
>     ____sep =  np.degrees(target.separation(pot_cal,antenna=ant))
>     ____print "%s, is %4.4f Degrees from %s "%(target.name ,sep,pot_cal.name)
> 
>

**Comment by mserylak** [comment]
We should get Tony to make some order in katsdpcatalogues before we merge anything to katconfig. Besides, katconfig also has gazillion branches last time I checked...

**Comment by ajoubertza** [comment]
Some CAM operational notes which might help your decision:
- katconfig repo:
  - There is only one copy of this checked out, on our so-called "head node".  The catalogues are available in observations via `kat.sources` (a katpoint catalogue).
  - On site we always keep this on the _karoo_ branch
  - There is a cron job that automatically updates it (git pull) every 15 min or so.
  - The git hash of this repo is part of the "config label" that is recorded (and can be chosen) when activating a subarray.  Note that once the subarray is activated, all observations in that subarray build will use the same config label (the instruction set gets a `--config-label` argument).  I.e. even if the cron job updates the repo while the subarray is active, this will not change what is used for the observations.  Only on the next subarray activation will the new changes come in.
- katsdpscripts repo:
  - there are many copies of this checked out, but the one on the obs node is that one that will be used.
  - this runs on the _master_ branch.
  - it is not updated automatically.
  - changes here will be available immediately the next time an observation tries to read one of the files.

**Comment by rubyvanrooyen** [comment]
With advice from Sharmila a move has started to get the guaranteed calibrators into katconfig
Jira: https://skaafrica.atlassian.net/browse/SR-1354
Pull request: https://github.com/ska-sa/katconfig/pull/780

Things to immediately chat about when the pull request is done
Updating calibration observation scripts such as phase-up and delay-calibration to point to the katconfig catalogues

**Comment by rubyvanrooyen** [comment]
Closing this issue, follow in the katconfig pull request
